src/nanodbc.cpp

```diff
@@ -994,15 +994,15 @@ class connection::connection_impl
         return string_type(&name[0], &name[strarrlen(name)]);
     }
 
-
     std::size_t ref_transaction()
     {
-        return --transactions_;
+        return ++transactions_;
     }
 
     std::size_t unref_transaction()
     {
-        return ++transactions_;
+        NANODBC_ASSERT(transactions_ != 0);
```

Comment:
Is there a benefit to not asserting here? Perhaps if `transactions_` already equals `0` we can just return?